### PR TITLE
Initial project setup and splash screen

### PR DESCRIPTION
**Pull request by edwin-pixel-1**
Creation of basic project and implementation of splash screen with an image animation and the transition to the empty subscribers list activity with delay of 3.0 seconds.

### CODE REVIEW

File: app/src/main/java/com/edwincobos/subscribersapp/commons/abstracts/AbstractActivity.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.edwincobos.subscribersapp.commons.abstracts;
+
+import android.support.v7.app.AppCompatActivity;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/abstracts/AbstractActivity.java]: Please use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/commons/models/ItemUserList.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package com.edwincobos.subscribersapp.commons.models;
+
+import java.util.ArrayList;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/models/ItemUserList.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/commons/models/ItemUserList.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ */
+public class ItemUserList {
+
+    private int id;
+
+    //@SerializedName("login")
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/models/ItemUserList.java]: Do not upload commented code. If it's not used it, do not commit it.

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import com.edwincobos.subscribersapp.commons.models.ItemUserList;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.http.GET;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                    .build();
+
+            apiService = retrofit.create(GithubApiInterface.class);
+        }
+
+        return apiService;
+    }
+
+    /**
+     * This interface define the end-points of service
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java]: Add the misses "s": defines.

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
+ */
+public class ApiClientGithub {
+    private static GithubApiInterface apiService;
+
+    public static GithubApiInterface getApiService() {
+
+        if(apiService == null){
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java]: Check code format, you missed some spaces.

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Constants.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.edwincobos.subscribersapp.commons.utils;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Constants.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Utils.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package com.edwincobos.subscribersapp.commons.utils;
+
+import android.util.Log;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Utils.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Constants.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+ */
+public class Constants {
+
+    public static final String GITHUB_BASE_URL = "https://api.github.com/";
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Constants.java]: :+1:

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 10]
```diff
@@ -1,34 +1,33 @@
...
+import android.widget.ImageView;
 
 import com.edwincobos.subscribersapp.R;
+import com.edwincobos.subscribersapp.commons.abstracts.AbstractActivity;
 import com.edwincobos.subscribersapp.subscriberslist.SubscribersListActivity;
 
-import pl.droidsonroids.gif.GifImageView;
-
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AbstractActivity {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java]: Please add some javadoc.

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 11]
```diff
@@ -1,34 +1,33 @@
...
     private void startAnimation() {
-        gifImageView = (GifImageView) findViewById(R.id.splashLogo);
+        logoImage = (ImageView) findViewById(R.id.splashLogo);
 
-        gifImageView.setScaleX(0.0f);
-        gifImageView.setScaleY(0.0f);
-        gifImageView.animate().setStartDelay(200).setDuration(1300).scaleX(1.0f).scaleY(1.0f);
+        logoImage.setScaleX(0.0f);
+        logoImage.setScaleY(0.0f);
+        logoImage.animate().setStartDelay(200).setDuration(1300).scaleX(1.0f).scaleY(1.0f);
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java]: Please, define constants for these values

File: .idea/.name [hunk 12]
```diff
@@ -0,0 +1 @@
+Git-Subscriber-List-Dev
```
Comment (daniellemariani) [inline .idea/.name]: Please add .idea to gitignore

File: .idea/compiler.xml [hunk 13]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daniellemariani) [inline .idea/compiler.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/copyright/profiles_settings.xml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
```
Comment (daniellemariani) [inline .idea/copyright/profiles_settings.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/encodings.xml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daniellemariani) [inline .idea/encodings.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/gradle.xml [hunk 16]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daniellemariani) [inline .idea/gradle.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/misc.xml [hunk 17]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daniellemariani) [inline .idea/misc.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/modules.xml [hunk 18]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (daniellemariani) [inline .idea/modules.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/runConfigurations.xml [hunk 19]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (daniellemariani) [inline .idea/runConfigurations.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: .idea/vcs.xml [hunk 20]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (daniellemariani) [inline .idea/vcs.xml]: Same here, this should not be commited. Please add .idea to gitignore

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.http.GET;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
+ */
+public class ApiClientGithub {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/ApiClientGithub.java]: I thought you will send this in another PR

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.widget.ImageView;
+
+import com.edwincobos.subscribersapp.R;
+import com.edwincobos.subscribersapp.commons.abstracts.AbstractActivity;
+import com.edwincobos.subscribersapp.subscriberslist.SubscribersListActivity;
+
+public class MainActivity extends AbstractActivity {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java]: Add some javadoc

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+        startAnimation();
+        startTimer();
+    }
+
+    private void startAnimation() {
+        logoImage = (ImageView) findViewById(R.id.splashLogo);
+
+        logoImage.setScaleX(0.0f);
+        logoImage.setScaleY(0.0f);
+        logoImage.animate().setStartDelay(200).setDuration(1300).scaleX(1.0f).scaleY(1.0f);
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java]: Define constants please

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    private void startTimer(){
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                Intent intent = new Intent();
+                intent.setClass(getApplicationContext(), SubscribersListActivity.class);
+                startActivity(intent);
+                //overridePendingTransition(R.anim.effect_fade_in, R.anim.effect_fade_out);
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java]: Do not commit commented code

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+                finish();
+            }
+        }, 3000);
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java]: Define it as a constant

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java [hunk 26]
```diff
@@ -0,0 +1,39 @@
+package com.edwincobos.subscribersapp.subscriberslist;
+
+import android.os.Bundle;
+
+import com.edwincobos.subscribersapp.R;
+import com.edwincobos.subscribersapp.commons.abstracts.AbstractActivity;
+
+public class SubscribersListActivity extends AbstractActivity implements SubscribersListView {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java]: Add some javadoc

Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java]: I thought this will be added in another PR

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListInteractor.java [hunk 27]
```diff
@@ -0,0 +1,17 @@
+package com.edwincobos.subscribersapp.subscriberslist;
+
+import com.edwincobos.subscribersapp.commons.models.ItemUserList;
+
+import java.util.List;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListInteractor.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListInteractor.java [hunk 28]
```diff
@@ -0,0 +1,17 @@
...
+ */
+public interface SubscribersListInteractor {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListInteractor.java]: I thought this would be in another PR

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListInteractorImp.java [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+import com.edwincobos.subscribersapp.commons.utils.Utils;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListInteractorImp.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListPresenter.java [hunk 30]
```diff
@@ -0,0 +1,12 @@
+package com.edwincobos.subscribersapp.subscriberslist;
+
+/**
+ * Created by edwin.cobos on 18/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListPresenter.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListPresenterImp.java [hunk 31]
```diff
@@ -0,0 +1,38 @@
+package com.edwincobos.subscribersapp.subscriberslist;
+
+import com.edwincobos.subscribersapp.commons.models.ItemUserList;
+import com.edwincobos.subscribersapp.commons.utils.Utils;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by edwin.cobos on 18/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListPresenterImp.java]: Use @author annotation

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListPresenterImp.java [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+public class SubscribersListPresenterImp implements SubscribersListPresenter, SubscribersListInteractor.OnFinishedListener {
+
+    private SubscribersListView view;
+    private SubscribersListInteractor interactor;
+
+    public SubscribersListPresenterImp(SubscribersListView view){
+        this.view = view;
+        this.interactor = new SubscribersListInteractorImp();
+
+        //interactor.getSubscribersDataList(this);
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListPresenterImp.java]: Do not commit commented code

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListView.java [hunk 33]
```diff
@@ -0,0 +1,14 @@
+package com.edwincobos.subscribersapp.subscriberslist;
+
+/**
+ * Created by edwin.cobos on 19/08/2016.
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListView.java]: Use @author annotation

File: app/src/main/res/layout/activity_main.xml [hunk 34]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".main.MainActivity"
+    android:background="#ffffff">
```
Comment (daniellemariani) [inline app/src/main/res/layout/activity_main.xml]: Use colors.xml

File: app/src/main/res/layout/activity_main.xml [hunk 35]
```diff
@@ -0,0 +1,32 @@
...
+
+    <TextView
+        android:gravity="center_horizontal"
+        android:padding="16dp"
```
Comment (daniellemariani) [inline app/src/main/res/layout/activity_main.xml]: Use dimens.xml

File: app/src/main/res/layout/activity_main.xml [hunk 36]
```diff
@@ -0,0 +1,32 @@
...
+        android:textSize="25sp"
```
Comment (daniellemariani) [inline app/src/main/res/layout/activity_main.xml]: Same here

**Comment by daniellemariani** [comment]
Please, split this PR at least in two. You are adding 994 new lines when it should be a minor number so as to make simple the CR.

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java (hunk 26 above)
Comment (edwin-pixel-1) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java]: Sorry for the mistake, I thought the PR sent doesn't update from new commits. 
To the future, if there is a PR open I'm going to create a new branch for new commits

File: app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Constants.java [hunk 37]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * @author edwin.cobos
+ * @since 23/08/2016
+ */
+public class Constants {
+
+    public static final long SPLASH_DELAY = 3000;
+
+    public final static class SPLASH_LOGO_ANIMATION{
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/commons/utils/Constants.java]: Check code format. Missing space before the brace

File: app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java [hunk 38]
```diff
@@ -0,0 +1,50 @@
...
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.ImageView;
+
+import com.edwincobos.subscribersapp.R;
+import com.edwincobos.subscribersapp.commons.utils.Constants;
+import com.edwincobos.subscribersapp.subscriberslist.SubscribersListActivity;
+
+public class MainActivity extends AppCompatActivity {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/main/MainActivity.java:17]: Add javadoc

File: app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java [hunk 39]
```diff
@@ -0,0 +1,15 @@
+package com.edwincobos.subscribersapp.subscriberslist;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+
+import com.edwincobos.subscribersapp.R;
+
+public class SubscribersListActivity extends AppCompatActivity {
```
Comment (daniellemariani) [inline app/src/main/java/com/edwincobos/subscribersapp/subscriberslist/SubscribersListActivity.java]: Do we really need this class in this PR?

File: app/src/main/res/values/dimens.xml [hunk 40]
```diff
@@ -0,0 +1,7 @@
+<resources>
+    <!-- Default screen margins, per the Android Design guidelines. -->
+    <dimen name="activity_horizontal_margin">16dp</dimen>
+    <dimen name="activity_vertical_margin">16dp</dimen>
+    <dimen name="splash.title.size">25sp</dimen>
```
Comment (daniellemariani) [inline app/src/main/res/values/dimens.xml]: Please use "_" to separate the words

File: app/src/main/res/values/strings.xml [hunk 41]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="app_name">Github Subscribers</string>
+    <string name="splash_title">GITHUB SUBSCRIBER</string>
```
Comment (daniellemariani) [inline app/src/main/res/values/strings.xml]: Did you miss the "S" at the end?

File: app/src/main/res/values/colors.xml [hunk 42]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="colorPrimary">#3F51B5</color>
+    <color name="colorPrimaryDark">#303F9F</color>
+    <color name="colorAccent">#FF4081</color>
+    <color name="colorTitle">#000</color>
+    <color name="backgroundWhite">#FFFFFF</color>
```
Comment (daniellemariani) [inline app/src/main/res/values/colors.xml]: For new color resources let's use "_" to separate the words: background_white (This could be called white, so it matches for everthing, text fonts, backgrounds, etc.)

File: app/src/main/res/layout/activity_main.xml [hunk 43]
```diff
@@ -0,0 +1,32 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".main.MainActivity"
+    android:background="@color/white">
+
+    <TextView
+        android:gravity="center_horizontal"
+        android:padding="@dimen/activity_horizontal_margin"
+        android:textSize="@dimen/splash.title.size"
```
Comment (daniellemariani) [inline app/src/main/res/layout/activity_main.xml]: I think this resource was renamed.

**Comment by daniellemariani** [comment]
LG
